Makefile

```diff
@@ -28,27 +30,49 @@ test: fmt vet ## Run tests.
 	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
 	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./... -coverprofile cover.out -shuffle on
 
+test-e2e:
+	./scripts/deploy-on-kind.sh e2e
+	KUBECONFIG="${HOME}/.kube/e2e.yml" API_SERVER_ROOT=http://localhost ROOT_NAMESPACE=cf-k8s-api-system go test -tags e2e -count 1 ./tests/e2e
+
 run: fmt vet ## Run a controller from your host.
 	go run ./main.go
 
 generate: fmt vet
 	go generate ./...
 
 deploy: kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
-	$(KUSTOMIZE) build config | kubectl apply -f -
+	cd config/base && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
+	$(KUSTOMIZE) build config/base | kubectl apply -f -
+
+deploy-kind: kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd config/base && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
+	$(KUSTOMIZE) build config/overlays/kind | kubectl apply -f -
 
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config | kubectl delete -f -
+	$(KUSTOMIZE) build config/base | kubectl delete -f -
+
+docker-build: ## Build docker image with the manager.
+	docker build -t ${IMG} .
 
 build-reference: kustomize
-	cd config && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
-	$(KUSTOMIZE) build config -o reference/cf-k8s-api.yaml
+	cd config/base && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
+	$(KUSTOMIZE) build config/base -o reference/cf-k8s-api.yaml
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
 	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.2.0)
 
+HNC_VERSION ?= v0.8.0
+hnc-install:
+	kubectl label ns kube-system hnc.x-k8s.io/excluded-namespace=true --overwrite
+	kubectl label ns kube-public hnc.x-k8s.io/excluded-namespace=true --overwrite
+	kubectl label ns kube-node-lease hnc.x-k8s.io/excluded-namespace=true --overwrite
+	kubectl apply -f https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-$(HNC_VERSION)/hnc-manager.yaml
+	kubectl rollout status deployment/hnc-controller-manager -w -n hnc-system
+	echo -n waiting for manager to be ready and servicing validating webhooks
+	until kubectl logs -n hnc-system deployment/hnc-controller-manager manager | grep -q "setup complete"; do echo -n .; sleep 0.5; done
```

Comment:
This is an extremely brittle check - is there another way we can check this in the future?

Reply:
yep - this was addressed in the list spaces PR: https://github.com/cloudfoundry/cf-k8s-api/commit/1a66502b5fdd7b951f5f3842ce874121822688be